common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/goal/FollowOwnerGoal.java

```diff
@@ -0,0 +1,138 @@
+package com.github.teamfusion.summonerscrolls.common.entity.goal;
+
+import com.github.teamfusion.summonerscrolls.common.entity.ISummon;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.ai.goal.Goal;
+import net.minecraft.world.entity.ai.navigation.PathNavigation;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.LeavesBlock;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.pathfinder.BlockPathTypes;
+import net.minecraft.world.level.pathfinder.WalkNodeEvaluator;
+
+import java.util.EnumSet;
+
+public class FollowOwnerGoal extends Goal {
```

Comment:
Doesn't this already exist as a Vanilla goal?

Reply:
It does but since the summons doesn't extend the tameable mob I had to make my own.